services/galley/src/Galley/Data.hs

```diff
@@ -183,10 +184,23 @@ teamConversations t =
   map (uncurry newTeamConversation)
     <$> retry x1 (query Cql.selectTeamConvs (params Quorum (Identity t)))
 
-teamMembers :: forall m. (MonadThrow m, MonadClient m) => TeamId -> m [TeamMember]
-teamMembers t =
+teamMembers :: forall m. (MonadThrow m, MonadClient m) => TeamId -> Range 1 2000 Int32 -> m [TeamMember]
+teamMembers t limit =
   mapM newTeamMember'
-    =<< retry x1 (query Cql.selectTeamMembers (params Quorum (Identity t)))
+    =<< retry x1 (query Cql.selectTeamMembers (params Quorum (t, limit)))
+  where
+    newTeamMember' ::
+      (UserId, Permissions, Maybe UserId, Maybe UTCTimeMillis, Maybe UserLegalHoldStatus) ->
+      m TeamMember
+    newTeamMember' (uid, perms, minvu, minvt, mlhStatus) =
+      newTeamMemberRaw uid perms minvu minvt (fromMaybe UserLegalHoldDisabled mlhStatus)
+
+-- | This operation gets **all** members of a team, this should go away before we
+-- roll out large teams
```

Comment:
```suggestion
-- | TODO: This operation gets **all** members of a team, this should go away before
-- we roll out large teams
```

(there really was no need for changing the linebreak, but i wanted to try out multi-line edits in the github ui.  :))